framework/src/fviks/FVInterfaceKernel.C

```diff
@@ -0,0 +1,161 @@
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVInterfaceKernel.h"
+#include "FEProblemBase.h"
+#include "SystemBase.h"
+#include "MooseVariableFV.h"
+#include "Assembly.h"
+
+InputParameters
+FVInterfaceKernel::validParams()
+{
+  InputParameters params = MooseObject::validParams();
+  params += TransientInterface::validParams();
+  params += BoundaryRestrictableRequired::validParams();
+  params += TaggingInterface::validParams();
+
+  params.addRequiredParam<std::vector<SubdomainName>>(
+      "subdomain1", "The subdomains on the 1st side of the boundary.");
+  params.addRequiredParam<std::vector<SubdomainName>>(
+      "subdomain2", "The subdomains on the 2nd side of the boundary.");
+  params.addRequiredParam<NonlinearVariableName>(
+      "variable1", "The name of the first variable that this interface kernel applies to");
+  params.addParam<NonlinearVariableName>(
+      "variable2",
+      "The name of the second variable that this interface kernel applies to. If not supplied, "
+      "variable1 will be used.");
+  params.addParam<bool>("use_displaced_mesh",
+                        false,
+                        "Whether or not this object should use the "
+                        "displaced mesh for computation.  Note that "
+                        "in the case this is true but no "
+                        "displacements are provided in the Mesh block "
+                        "the undisplaced mesh will still be used.");
+
+  params.addParamNamesToGroup("use_displaced_mesh", "Advanced");
+  params.addCoupledVar("displacements", "The displacements");
+  params.declareControllable("enable");
+  params.registerBase("FVInterfaceKernel");
+  params.registerSystemAttributeName("FVInterfaceKernel");
+  return params;
+}
+
+FVInterfaceKernel::FVInterfaceKernel(const InputParameters & parameters)
+  : MooseObject(parameters),
+    BoundaryRestrictableRequired(this, false),
+    SetupInterface(this),
+    FunctionInterface(this),
+    DistributionInterface(this),
+    UserObjectInterface(this),
+    TransientInterface(this),
+    PostprocessorInterface(this),
+    VectorPostprocessorInterface(this),
+    GeometricSearchInterface(this),
+    Restartable(this, "BCs"),
+    MeshChangedInterface(parameters),
+    TaggingInterface(this),
+    NeighborCoupleableMooseVariableDependencyIntermediateInterface(
+        this, /*nodal=*/false, /*neighbor_nodal=*/false, /*is_fv=*/true),
+    TwoMaterialPropertyInterface(this, Moose::EMPTY_BLOCK_IDS, boundaryIDs()),
+    _tid(getParam<THREAD_ID>("_tid")),
+    _sys(*getCheckedPointerParam<SystemBase *>("_sys")),
+    _subproblem(*getCheckedPointerParam<SubProblem *>("_subproblem")),
+    _nonconst_var1(_sys.getFVVariable<Real>(_tid, getParam<NonlinearVariableName>("variable1"))),
+    _nonconst_var2(_sys.getFVVariable<Real>(_tid,
```

Comment:
why the non const versions?

Reply:
We need the non-const versions for the calls to `addMooseVariableDependency`. Honestly the reason that that method takes non-const variables is lost from my memory...I think there's a good reason for it...but I wouldn't stake my life on it

Reply:
But I made these members private to prevent abuse in derived classes